packages/typespec-client-generator-core/generated-defs/Azure.ClientGenerator.Core.ts

```diff
@@ -433,6 +433,36 @@ export type UseSystemTextJsonConverterDecorator = (
   scope?: string
 ) => void;
 
+/**
+ * Client parameters you would like to add to the client. By default, we apply endpoint, credential, and api-version parameters. If you add clientInitialization, we will append those to the default list of parameters.
+ *
+ * @param scope The language scope you want this decorator to apply to. If not specified, will apply to all language emitters
+ * @example
+ * ```typespec
+ * // main.tsp
+ * namespace MyService;
+ *
+ * op upload(blobName: string): void;
+ * op download(blobName: string): void;
+ *
+ * // client.tsp
+ * namespace MyCustomizations;
+ * model MyServiceClientOptions {
+ *   blobName: string;
+ * }
+ *
+ * @@clientInitialization(MyService, MyServiceClientOptions)
+ * // The generated client will have `blobName` on it. We will also
+ * // elevate the existing `blobName` parameter to the client level.
+ * ```
+ */
+export type ClientInitializationDecorator = (
+  context: DecoratorContext,
+  target: Namespace | Interface,
+  options: Model,
```

Comment:
is it possible that we have a consolidated decorator signature like `@override` if we want to make this decorator flexible like redefine the initialization?

Reply:
I'm not quite sure what you're suggesting, are you suggesting we use `@override` instead of `@clientInitialization`? If so, in some ways they're similar, but I feel since there is no original "client initialization" that is defined in the tsp, using the decorator `@override` wouldn't make sense to users